desc/grid.py

```diff
@@ -515,14 +516,32 @@ class Grid(_Grid):
         etc may be wrong if grid contains duplicate nodes.
     """
 
-    def __init__(self, nodes, sort=False, jitable=False, **kwargs):
+    def __init__(
+        self, nodes, spacing=None, weights=None, sort=False, jitable=False, **kwargs
+    ):
         # Python 3.3 (PEP 412) introduced key-sharing dictionaries.
         # This change measurably reduces memory usage of objects that
         # define all attributes in their __init__ method.
         self._NFP = 1
         self._sym = False
         self._node_pattern = "custom"
-        self._nodes, self._spacing = self._create_nodes(nodes)
+        self._nodes = self._create_nodes(nodes)
+        if spacing is not None:
+            spacing = (
+                jnp.atleast_2d(jnp.asarray(spacing))
+                .reshape(self.nodes.shape)
+                .astype(float)
+            )
+            self._spacing = spacing
+        if weights is None and spacing is not None:
+            self._weights = self._spacing.prod(axis=1)
```

Comment:
it seems like you're making an effort to ensure jax arrays. maybe because it's useful for [memory](https://github.com/PlasmaControl/DESC/blob/90d1ecb2599cb1127f09c838073e679a4dafbb40/desc/compute/utils.py#L924). in that case this should be jnp.prod(self.spacing, axis=1)

Reply:
at this point `self._spacing` will already be a jax array so I think they should be the same.